Nautilus/Utility/ThunderkitUtilities/ApplySNLayer.cs

```diff
@@ -0,0 +1,75 @@
+using System.Reflection;
+using UnityEngine;
+
+namespace Nautilus.Utility.ThunderkitUtilities;
+
+internal class ApplySNLayer : MonoBehaviour
+{
+    [Tooltip("The name of the layer you want to apply")]
+    public LayerName layerName;
+
+    [Tooltip("How to apply the layer")]
+    public GeneralSetMode layerSetMode;
```

Comment:
Should this be GeneralSetMode or MaterialSetMode? Feels odd to not allow the ui only option, when the materials have it

Reply:
I can't really see a use case where the UI only would be needed but I suppose there's no harm in adding it